src/main/java/uk/gov/ons/census/casesvc/service/CaseReceiptService.java

```diff
@@ -1,27 +1,60 @@
 package uk.gov.ons.census.casesvc.service;
 
+import static uk.gov.ons.census.casesvc.utility.QuestionnaireTypeHelper.isIndividualQuestionnaireType;
 import static uk.gov.ons.census.casesvc.utility.QuestionnaireTypeHelper.iscontinuationQuestionnaireTypes;
 
+import java.util.Optional;
 import org.springframework.stereotype.Component;
 import uk.gov.ons.census.casesvc.model.entity.Case;
 import uk.gov.ons.census.casesvc.model.entity.UacQidLink;
+import uk.gov.ons.census.casesvc.model.repository.CaseRepository;
 
 @Component
 public class CaseReceiptService {
   private final CaseService caseService;
+  private final CaseRepository caseRepository;
 
-  public CaseReceiptService(CaseService caseService) {
+  public CaseReceiptService(CaseService caseService, CaseRepository caseRepository) {
     this.caseService = caseService;
+    this.caseRepository = caseRepository;
   }
 
   public void receiptCase(UacQidLink uacQidLink) {
     Case caze = uacQidLink.getCaze();
 
     if (caze.isReceiptReceived()) return;
 
-    if (!iscontinuationQuestionnaireTypes(uacQidLink.getQid())) {
-      caze.setReceiptReceived(true);
-      caseService.saveAndEmitCaseUpdatedEvent(caze);
+    if (iscontinuationQuestionnaireTypes(uacQidLink.getQid())) return;
+
+    if (caze.getCaseType().equals("CE") && isIndividualQuestionnaireType(uacQidLink.getQid())) {
+      incrementActualResponseAndSetReceiptedIfAppropriate(caze);
+      return;
+    }
+
+    caze.setReceiptReceived(true);
+    caseService.saveAndEmitCaseUpdatedEvent(caze);
+  }
+
+  private void incrementActualResponseAndSetReceiptedIfAppropriate(Case caze) {
+    /*
+      This stops the actualResponses being updated by another thread for another receipt or linking event
+    */
+    Optional<Case> oCase = caseRepository.getCaseAndLockByCaseId(caze.getCaseId());
+
+    if (!oCase.isPresent()) {
+      throw new RuntimeException(
+          "Failed to get row to increment responses, row is probably locked and this should resolve itself: "
+              + caze.getCaseId());
     }
+
+    Case lockedCase = oCase.get();
+    lockedCase.setCeActualResponses(lockedCase.getCeActualResponses() + 1);
+
+    if (lockedCase.getAddressLevel().equals("U")
+        && lockedCase.getCeActualResponses().compareTo(lockedCase.getCeExpectedCapacity()) >= 0) {
```

Comment:
I'm really not a fan of `compareTo` unless you're comparing strings, which you're not. What's wrong with good old fashioned `>=`?

Reply:
Changed it to use intValues and >=